### PR TITLE
Improve HTTP/CoAP C&C tracing

### PR DESCRIPTION
**Pull request by calohmn**
This adds a `FOLLOWS_FROM` span reference from the HTTP/CoAP ttd request trace to the command-handling trace started in the Command Router.
Also decouples the closing of the command consumer from the request handling trace spans by means of an extra `FOLLOWS_FROM` span, the motivation being that the request handling trace span ("POST"/"PUT") gets finished before the span for closing the command consumer anyway.

Before:
HttpSpan_Before

With this PR:
HttpSpan_link_to_CmdRouter

### CODE REVIEW

File: adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/AbstractHonoResource.java [hunk 1]
```diff
@@ -562,6 +571,7 @@ protected void addCommandToResponse(
                         commandSample);
                 LOG.debug("command message is invalid: {}", command);
                 commandContext.reject("malformed command message");
+                processCommandSpan.finish();
```
Comment (sophokles73) [inline adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/AbstractHonoResource.java:574]: can't we simply pull this out of the top level *if* statement?

Comment (calohmn) [reply adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/AbstractHonoResource.java:574]: Not really - the first of the 3 `processCommandSpan.finish()` invocations (in line 548) is inside an `onComplete` block, so it's invoked some time later.

Comment (sophokles73) [reply adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/AbstractHonoResource.java:574]: oh, right

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**